crates/boundless-market/build.rs

```diff
@@ -193,17 +193,31 @@ fn copy_interfaces_and_types() {
     }
 }
 
-fn copy_artifacts() {
+fn generate_contracts_rust_file() {
     println!("cargo::rerun-if-env-changed=CARGO_CFG_TARGET_OS");
     let manifest_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    let dest_path = Path::new(&manifest_dir).join("src/contracts/artifacts");
-    fs::create_dir_all(&dest_path).unwrap();
-
     let src_path =
         Path::new(&manifest_dir).parent().unwrap().parent().unwrap().join("contracts").join("out");
 
+    // Check if we're running under cargo publish
+    let is_cargo_publish = std::env::var("CARGO_RELEASE_UPLOAD").is_ok();
+
+    // If running under cargo publish and the contracts directory doesn't exist,
+    // we should exit early rather than fail trying to find contract JSON files
+    if is_cargo_publish && !src_path.exists() {
+        println!("cargo:warning=Skipping contract bytecode generation during cargo publish");
+        return;
+    }
+
+    // Start with file header content
+    let mut rust_content = String::from("// Auto-generated file, do not edit manually\n\n");
+
     for contract in ARTIFACT_TARGET_CONTRACTS {
         let source_path = src_path.join(format!("{contract}.sol/{contract}.json"));
+        if !source_path.exists() {
+            println!("cargo:warning=Skipping contract bytecode generation");
+            return;
+        }
```

Comment:
When a single file doesn't exist, essentially nothing gets written no default is used?

Reply:
Yeah although that I just a trick for cargo publish, as otherwise the one checked on GitHub will be used